TestsFoundation/TestsFoundation/Extensions/XCUIElementExtensions.swift

```diff
@@ -19,50 +19,241 @@
+    }
+
+    func tapAt(_ point: CGPoint) {
+        waitUntil(condition: .hittable)
+        coordinate(withNormalizedOffset: .zero).withOffset(CGVector(dx: point.x, dy: point.y)).tap()
+    }
+
+    // MARK: Find
+    func find(label: String, type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).matching(label: label).firstMatch
     }
 
-    func find(labelContaining needle: String, type: XCUIElement.ElementType = .any) -> Element {
-        return descendants(matching: type).matching(labelContaining: needle).firstElement
+    func find(labelContaining needle: String, type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).matching(labelContaining: needle).firstMatch
     }
 
-    func find(id: String, type: XCUIElement.ElementType = .any) -> Element {
-        return descendants(matching: type).matching(id: id).firstElement
+    func find(id: String, type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).matching(id: id).firstMatch
     }
 
-    func find(idStartingWith prefix: String, type: XCUIElement.ElementType = .any) -> Element {
-        return descendants(matching: type).matching(idStartingWith: prefix).firstElement
+    func find(idStartingWith prefix: String, type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).matching(idStartingWith: prefix).firstMatch
     }
 
-    func find(value: String, type: XCUIElement.ElementType = .any) -> Element {
-        return descendants(matching: type).matching(value: value).firstElement
+    func find(value: String, type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).matching(value: value).firstMatch
     }
 
-    func find(type: XCUIElement.ElementType, index: Int = 0) -> Element {
-        return XCUIElementQueryWrapper(descendants(matching: type), index: index)
+    func find(type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).firstMatch
     }
 
-    func find(id: String, label: String, type: XCUIElement.ElementType = .any) -> Element {
-        return descendants(matching: type).matching(id: id).matching(label: label).firstElement
+    func find(id: String, label: String, type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).matching(id: id).matching(label: label).firstMatch
     }
 
-    func findAll(type: XCUIElement.ElementType) -> [Element] {
-        XCUIElementQueryWrapper(descendants(matching: type)).allElements
+    func findAll(type: XCUIElement.ElementType, minimumCount: Int = 1, timeout: TimeInterval = defaultTimeout, gracePeriod: UInt32 = defaultGracePeriod) -> [XCUIElement] {
+        let deadline = Date().addingTimeInterval(timeout)
+        var result = descendants(matching: type).allElementsBoundByIndex
+        while Date() < deadline {
+            if result.count >= minimumCount { return result } else {
```

Comment:
This formatting is weird to me since I expect that statements on the next line following the condition will be executed if the condition evaluates to true. Let's move the else branch to a new line or use a guard for that early exit.

Reply:
Done.